Model/Processor/ImportProcessor.php

```diff
@@ -85,8 +85,18 @@ public function __construct(
     public function process()
     {
         $files = $this->finder->find();
-        if (0 === count($files)) {
-            throw new \InvalidArgumentException('No files found for format: *.' . $this->getFormat());
+
+        try {
+            if (0 === count($files)) {
+                throw new \InvalidArgumentException('No files found for format: *.' . $this->getFormat());
+            }
+        } catch (\InvalidArgumentException $exception) {
+            $inputOptions = $this->getInput()->getOptions();
+            if (true === $this->getInput()->getOption('allow-empty-directories')) {
+                $this->getOutput()->writeln('No files found for format: *.' . $this->getFormat());
+            } else {
+                throw $exception;
+            }
```

Comment:
We can simplify the behavior instead of catching the exception and then re-throwing the exception. Also, an additional output message (`No files found...`) is unnecessary, in my opinion, since the new option `allow-empty-directories` will be used intentionally by developers. I would change it to something like this:

```suggestion
        if (0 === count($files) && false === $this->getInput()->getOption('allow-empty-directories')) {
            throw new \InvalidArgumentException('No files found for format: *.' . $this->getFormat());
        }
```

@torhoehn  What do you think?

Reply:
@therouv I had something like this before, but while looking at other code of this module I expected that this would be the preferred solution. I'll change that now.